includes/gateway-helpers.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * Calculate the profile start date to be sent to the payment gateway.
+ *
+ * @since TBD
+ *
+ * @param MemberOrder $order       The order to calculate the start date for.
+ * @param string      $date_format The format to use when formatting the profile start date.
+ * @param bool        $filter      Whether to filter the profile start date.
+ *
+ * @return string The profile start date in UTC time and the desired $date_format.
+ */
+function pmpro_calculate_profile_start_date( $order, $date_format, $filter = true ) {
+	// Calculate the profile start date.
+	$profile_start_date = date_i18n( 'Y-m-d H:i:s', strtotime( '+ ' . $order->BillingFrequency . ' ' . $order->BillingPeriod ) );
+
+	// Filter the profile start date if needed.
+	if ( $filter ) {
```

Comment:
Not sure if we'd totally need this parameter here. Do we have plans to ever not filter the profile start date when calling `pmpro_calculate_profile_start_date`?

Reply:
I'm currently hoping to use that param to determine if we can bundle initial/recurring payments when sending users to Stripe to check out:
https://github.com/strangerstudios/paid-memberships-pro/blob/415acf9cb02015d5310f9fa68d4ef2d4990b070d/classes/gateways/class.pmprogateway_stripe.php#L1657-L1669

The idea is that if the filtered profile start date is the same as the unfiltered profile start date, then no add on is changing the date of the first recurring payment so the subscription can be started immediately with the initial payment as a part of the subscription itself.

If there is another way to figure this out, though, then I would be ok removing that parameter.

Reply:
Ah got it, overlooked the $unfiltered_trial_period_days. 👍